File: core/src/main/java/com/garden/back/notification/utils/EmailUtils.java
package com.garden.back.notification.utils;

public final class EmailUtils {

    private EmailUtils() {
        throw new AssertionError("Utils class does not need to be initialized");
    }

    public static boolean isEmailAddress(Object potentiallyEmail) {
        if (potentiallyEmail instanceof String email) {
            return email.matches(REGEX);
        }
        return false;
    }

    public static String REGEX = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,7}$";
}

File: core/src/main/java/com/garden/back/notification/utils/EmailUtils.kt
package com.garden.back.notification.utils

object EmailUtils {
    fun isEmailAddress(potentiallyEmail: Any?): Boolean {
        val email = potentiallyEmail as? String ?: return false
        return email.matches(Regex(EMAIL_REGEX))
    }

    private const val EMAIL_REGEX = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,7}$"
}
